### ISSUE TITLE
Switching between similarly named queries fails to update query text area

### ISSUE THREAD

**Issue by KasperZutterman** [comunica/jQuery-Widget.js#87]
When switching between queries with similar names, the query text area does not update:
Replicate at http://comunica.kasperzutterman.com/

Replication steps:
- Select "Buurt in Gemeente" query
- Select "Buurt in Gemeente met @ reverse" query
After this, the "Buurt in Gemeente" query is still showing in the GrapQL Query text area
![Linked Data Fragments client - Google Chrome 29_06_2021 12_53_31](https://user-images.githubusercontent.com/21260838/123785756-31351f80-d8d9-11eb-90b5-f5e16009417d.png)

-----

## Bounty

A [bounty](https://comunica.dev/association/bounties/) has been placed on this issue by:

| |
|-|
| [<img alt="Comunica Association" src="https://comunica.dev/img/comunica_red.svg" height="50">](https://comunica.dev/association/) |
| €136 |

[Click here to learn more](https://comunica.dev/association/bounty_process/) if you're interested in claiming this bounty by resolving this issue.

**Comment by rubensworks** [comment]
I suspect this may have something to do with the fact that the query input fields are identical, but the context fields are not.

**Comment by rubensworks** [comment]
A bounty has been placed on this issue via the Comunica Association (see original post).

**Comment by milafrerichs** [comment]
> As discussed with @rubensworks, I will work on this issue.

**Comment by milafrerichs** [comment]
> I suspect this may have something to do with the fact that the query input fields are identical, but the context fields are not.

Yes, I can confirm, if the query is the same but the context is different, the context does not get updated.

**Comment by milafrerichs** [comment]
I have a solution which is probably not optimal but works and does not break anything.

```js
var queryContext = $queries.find(':selected').attr('queryContext');
if ((options.query !== $queries.val() || options.queryContext !== queryContext) && (query = $queries.val())) {
```

should I refactor to a better solution or is this good enough? :) if so, I'll create a PR.
